### PR TITLE
Support additional arguments for reducer

### PR DESCRIPTION
**Pull request by xiaohanzhang**
resolve acdlite/redux-actions/issues/148

### CODE REVIEW

**Comment by iofjuupasli** [comment]
@acdlite 
https://www.npmjs.com/package/combine-section-reducers also requires it

**Comment by NoumanSaleem** [comment]
@acdlite this is needed to support the "slice reducer" pattern outlined by redux: https://redux.js.org/docs/recipes/reducers/BeyondCombineReducers.html#sharing-data-between-slice-reducers

**Comment by xiaohanzhang** [comment]
@timche updated. 
I also added ```"trailingComma": "es5"``` to prettier setting, to fix conflict with lint

**Comment by timche** [comment]
@xiaohanzhang thanks for catching `trailingComma`. I've set it to `none` as we want to have it streamlined in all related repos. Please rebase again.

File: test/index.js [hunk 1]
```diff
@@ -20,3 +20,17 @@ test('chains multiple reducers into a single reducer', t => {
   t.deepEqual(reducerAddMult({ A: 1, B: 2 }, 3), { A: 12, B: 2 });
   t.deepEqual(reducerMultAdd({ A: 1, B: 2 }, 3), { A: 6, B: 2 });
 });
+
+test('should support additional arguments', t => {
```
Comment (timche) [inline test/index.js]: Please use `supports` instead of `should support`.

Comment (xiaohanzhang) [reply test/index.js]: fixed and rebased

**Comment by xiaohanzhang** [comment]
@timche fixed and rebased

**Comment by timche** [comment]
Thanks @xiaohanzhang, merged! :)

**Pull request merged**